src/main/java/org/openstreetmap/atlas/geography/converters/MultiplePolyLineToPolygonsConverterCommand.java

```diff
@@ -0,0 +1,85 @@
+package org.openstreetmap.atlas.geography.converters;
+
+import java.io.FileNotFoundException;
+import java.io.OutputStreamWriter;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.PolyLine;
+import org.openstreetmap.atlas.geography.Polygon;
+import org.openstreetmap.atlas.streaming.resource.File;
+import org.openstreetmap.atlas.streaming.writers.SafeBufferedWriter;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.openstreetmap.atlas.utilities.collections.StringList;
+import org.openstreetmap.atlas.utilities.conversion.StringConverter;
+import org.openstreetmap.atlas.utilities.runtime.Command;
+import org.openstreetmap.atlas.utilities.runtime.CommandMap;
+
+/**
+ * This command reads a file with delimited lists of {@link PolyLine}s in WKT format, and applies
+ * logic to stitch those {@link PolyLine}s together to form one or more {@link Polygon}s.
+ *
+ * @author matthieun
+ */
+public class MultiplePolyLineToPolygonsConverterCommand extends Command
+{
+    private static final Switch<File> POLYLINES = new Switch<>("polylines",
+            "File containing lines of semicolon separated list of WKT polylines", File::new,
+            Optionality.REQUIRED);
+    private static final Switch<File> POLYGONS = new Switch<>("polygons",
+            "Output file that will contain lines of semicolon separated list of reconstructed polygons",
+            File::new, Optionality.OPTIONAL);
+    private static final Switch<String> DELIMITER = new Switch<>("delimiter",
+            "The string delimiter between groups of polylines, and polygons in the output.",
+            StringConverter.IDENTITY, Optionality.OPTIONAL, ";");
+
+    private static final WktPolyLineConverter WKT_POLY_LINE_CONVERTER = new WktPolyLineConverter();
+    private static final MultiplePolyLineToPolygonsConverter MULTIPLE_POLY_LINE_TO_POLYGONS_CONVERTER = new MultiplePolyLineToPolygonsConverter();
+
+    public static void main(final String[] args)
+    {
+        new MultiplePolyLineToPolygonsConverterCommand().run(args);
+    }
+
+    @Override
+    protected int onRun(final CommandMap command)
+    {
+        final String delimiter = (String) command.get(DELIMITER);
+        final File inputFile = (File) command.get(POLYLINES);
+        final Iterable<List<PolyLine>> inputs = Iterables.stream(inputFile.lines())
+                .map(line -> StringList.split(line, delimiter).stream()
```

Comment:
How about using a flatMap instead of creating a map and then creating a stream of delimited string list again

Reply:
I believe that if I use flatmap here I would end up with one long list of polylines. What I am looking for here is each line in the input file to become a separate list of polylines. Not sure if that would work with a flatmap? If yes can you give a code example?

Reply:
Yes you are right. flatMap will flatten the iterable into long list of polylines. This way works better.